src/main/java/cart/domain/OrderManager.java

```diff
@@ -0,0 +1,64 @@
+package cart.domain;
+
+import cart.exception.PointException;
+import cart.exception.PriceInconsistencyException;
+
+import java.util.List;
+
+public class OrderManager {
+
+    private final List<CartItem> cartItems;
+
+    public OrderManager(List<CartItem> cartItems) {
+        this.cartItems = cartItems;
+    }
+
+    public void validateOrder(Order order) {
+        validateOriginalPrice(order.getOriginalPrice());
+        validateUsedPoint(order.getMember(), order.getUsedPoint());
+        validatePointToAdd(order.getPointToAdd());
+    }
+
+    private void validateOriginalPrice(Long originalPrice) {
+        int price = 0;
+        for (CartItem item : cartItems) {
+            price += item.getProduct().getPrice() * item.getQuantity();
+        }
+
+        if (originalPrice != price) {
+            throw new PriceInconsistencyException("주문 총액에 문제가 있습니다");
+        }
+    }
+
+    private void validateUsedPoint(Member member, Long usedPoint) {
+        if (member.getPoint() < usedPoint) {
+            throw new PointException.InvalidPointException("보유한 포인트보다 많은 포인트를 사용할 수 없습니다");
+        }
+
+        int price = 0;
+        for (CartItem item : cartItems) {
+            if (item.getProduct().getPointAvailable()) {
+                price += item.getProduct().getPrice();
+            }
+        }
+
+        if (price < usedPoint) {
+            throw new PointException.InvalidPointException("사용 가능한 포인트보다 많은 포인트를 사용할 수 없습니다");
+        }
+    }
+
+    private void validatePointToAdd(Long pointToAdd) {
+        double point = 0.0;
+        for (CartItem item : cartItems) {
+            point += item.getProduct().getPrice() * item.getQuantity() * item.getProduct().getPointRatio() / 100;
```

Comment:
```suggestion
            point += (item.getProduct().getPrice() * item.getQuantity() * item.getProduct().getPointRatio()) / 100;
```

확실히 연산 순서를 지키기 위해 괄호를 붙혀주는게 더 확실히 보이는거 같아요.

아울러 여기도 item이 해당 값을 리턴하도록 하는게 맞아보입니다.

Reply:
CartItem이 자신이 가진 물품의 가격, 적립되는 포인트, 포인트 사용가능 금액 등을 계산하도록 책임을 위임시켜주었고 OrderManager 내부에서는 cartItems를 stream을 통해 순회하도록 했습니다.

CartItem이 당연히 위의 책임들을 다 가지고 있는 게 자연스러운데 구현할 때는 떠올리지 못 했네요. 코멘트 감사드립니다!!